stellar_sdk/keypair.py

```diff
@@ -256,6 +258,54 @@ def from_mnemonic_phrase(
         )
         return cls.from_raw_ed25519_seed(raw_ed25519_seed)
 
+    @staticmethod
+    def generate_shamir_mnemonic_phrases(
+        member_threshold: int, member_count: int, passphrase: str = ""
+    ) -> list[str]:
+        """Generate mnemonic phrases using Shamir secret sharing method.
+
+        A randomly generated secret key is generated and split into `member_count`
+        mnemonic phrases. The secret key can be later reconstructed using any
+        subset of `member_threshold` phrases.
+
+        :param member_threshold: Number of members required to reconstruct the secret key.
+        :param member_count: Number of shares the secret is split into.
+        :param passphrase: An optional passphrase used to decrypt the secret key.
+        :return: A list of mnemonic phrases.
+        """
+        if member_count < member_threshold:
+            raise ValueError(
+                "There must be more members than the set threshold: "
+                "``member_count > member_threshold``."
+            )
+        secrets = Keypair.random().secret.encode()
+        return shamir_mnemonic.generate_mnemonics(
+            group_threshold=1,
+            groups=[(member_threshold, member_count)],
+            master_secret=secrets,
+            passphrase=passphrase.encode(),
+        )[0]
+
+    @classmethod
+    def from_shamir_mnemonic_phrases(
+        cls,
+        mnemonic_phrases: list[str],
+        passphrase: str = "",
+    ) -> "Keypair":
+        """Generate a :class:`Keypair` object via a list of mnemonic phrases.
+
+        :param mnemonic_phrases: A list of unique strings used to deterministically generate a keypair.
+        :param passphrase: An optional passphrase used to decrypt the secret key.
+        :return: A new :class:`Keypair` object derived from the mnemonic phrases.
+        """
+        try:
+            secret = shamir_mnemonic.combine_mnemonics(
+                mnemonics=mnemonic_phrases, passphrase=passphrase.encode()
+            )
+        except shamir_mnemonic.utils.MnemonicError as exc:
+            raise ValueError(exc) from exc
+        return cls.from_secret(secret.decode())
```

Comment:
What you have here is a master seed, and you cannot directly generate a KeyPair from it. Please refer to [SLIP-10](https://github.com/satoshilabs/slips/blob/master/slip-0010.md) for more information. 

Here is a fully functional example.
```python
import shamir_mnemonic
from stellar_sdk import Keypair

from stellar_sdk.sep.mnemonic import StellarMnemonic

# 2-of-3 Shamir Secret Sharing Scheme, generate from Trezor Safe 3
share0 = "glimpse buyer academic acid branch sled disaster sunlight material junction float emperor intend priority scene trash remember radar prospect dryer"
share1 = "glimpse buyer academic agency burden payroll alpha oven large amount smear forward pharmacy symbolic junk axle exercise segment frequent axle"
share2 = "glimpse buyer academic always careful become dance teaspoon daisy orange careful steady boundary exceed robin remind software grin space advocate"

master_seed = shamir_mnemonic.combine_mnemonics([share0, share2], passphrase=b"9012")

kp0 = Keypair.from_raw_ed25519_seed(StellarMnemonic.derive(master_seed, 0))  # m/44h/148h/0h
kp1 = Keypair.from_raw_ed25519_seed(StellarMnemonic.derive(master_seed, 1))
kp100 = Keypair.from_raw_ed25519_seed(StellarMnemonic.derive(master_seed, 100))

assert kp0.public_key == "GDZ4GYLVRLM2E6CGCOVYXAYMXJJAV3IHDXU6RUHX5AJVYS4AE6R6CHPJ"
assert kp1.public_key == "GCMZKXAAPQ3TDY5P7QDUVJBW66R2DGT7AM6MA3MCQENIF37E25U2PEK3"
assert kp100.public_key == "GCF7DAVTXXVQPOSB5TCA2CIFT7DZIPK23NCOV3RJ6FTYTZM6S6RPPACM"

```

Reply:
Ok, I have added the index support. I can use what you have here for a test afterwards.

Let me know if that's what you had in mind.